src/lib/statistics/generate_pruning_statistics.cpp

```diff
@@ -65,15 +73,15 @@ void generate_chunk_pruning_statistics(const std::shared_ptr<Chunk>& chunk) {
       } else {
```

Comment:
I am no fan of `// TODO(...)` comments, but can you please add one before line 69?
`// TODO(anyone): use dictionary-optimized path for FixedStringDictionarySegments as well.`

Maybe I can implement that tomorrow, but I am not sure.